soc/arm/infineon_cat1/psoc6/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2021 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_sources(soc.c)
+zephyr_include_directories(.)
+
+include(${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.psoc6_02.txt)
+
+# PSoC 6 family defines
+zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_CAT1A COMPONENT_CAT1A)
```

Comment:
Again, not sure if this belongs here or in the `modules/` folder

Reply:
I have same feeling....

Reply:
```
# PSoC 6 family defines
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1A COMPONENT_CAT1A)
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1 COMPONENT_CAT1)
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1 CY_USING_HAL)
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1 ${CONFIG_SOC})
```
moved to modules/hal_infineon